include/libtorrent/session_handle.hpp

```diff
@@ -449,6 +449,14 @@ namespace libtorrent
 		// posted, regardless of the alert mask.
 		void dht_live_nodes(sha1_hash const& nid);
 
+		// Query the DHT to retrieve a sample of the infohashes that the specific
+		// node currently have in their storage.
+		// The ``target`` is included for iterative lookups so that indexing nodes
+		// can perform a keyspace traversal with a single RPC per node by adjusting
+		// the target value for each RPC. It has no effect on the returned sample value.
+		// The result is posted as a ``dht_sample_infohashes_alert``.
+		void dht_sample_infohashes(udp::endpoint const& ep, sha1_hash const& target);
```

Comment:
If I understand correctly, while adjusting the `target`, nodes will return the field `nodes/nodes6` populated and the DHT will include those nodes. If after that, you use `dht_live_nodes` I expect that to be enough for the entire feature of "DHT indexing". Well, it could be a little convoluted, but the constraint is expose as little as possible in the public API.

`target` has no influence in `dht_sample_infohashes_alert::samples()`, the data returned by the nodes.

If it feels like I'm rephrasing the same, let me know :)